src/main/java/teammates/ui/controller/ImageUploadAction.java

```diff
@@ -0,0 +1,115 @@
+package teammates.ui.controller;
+
+import java.util.List;
+import java.util.Map;
+
+import teammates.common.util.AppUrl;
+import teammates.common.util.Config;
+import teammates.common.util.Const;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreFailureException;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+
+/**
+ * Action: uploads an image
+ */
+public class ImageUploadAction extends Action {
+
+    private FileUploadPageData data;
+
+    @Override
+    protected ActionResult execute() {
+        data = prepareData();
+
+        return createAjaxResult(data);
+    }
+
+    protected String getImageKeyParam() {
+        return Const.ParamsNames.IMAGE_TO_UPLOAD;
+    }
+
+    protected FileUploadPageData prepareData() {
```

Comment:
If this method is not overridden, consider making it `private` instead. Also applies for `extractImageKey`, `validateImage`, `deleteImage`, `deleteUploadedFile`.